lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php

```diff
@@ -868,16 +868,25 @@ protected function _getPortableTableIndexesList($tableIndexRows, $tableName=null
             $keyName = strtolower($keyName);
 
             if (!isset($result[$keyName])) {
+                $options = [
+                    'lengths' => [$tableIndex['sub_part'] ?? null],
+                ];
+
+                if (isset($tableIndex['where'])) {
+                    $options['where'] = $tableIndex['where'];
+                }
+
                 $result[$keyName] = [
                     'name' => $indexName,
                     'columns' => [$tableIndex['column_name']],
                     'unique' => $tableIndex['non_unique'] ? false : true,
                     'primary' => $tableIndex['primary'],
                     'flags' => $tableIndex['flags'] ?? [],
-                    'options' => isset($tableIndex['where']) ? ['where' => $tableIndex['where']] : [],
+                    'options' => $options,
                 ];
             } else {
                 $result[$keyName]['columns'][] = $tableIndex['column_name'];
+                $result[$keyName]['options']['lengths'][] = $tableIndex['sub_part'] ?? null;
```

Comment:
The same logic seems duplicated in `if` and `else ` branches. Can it be moved outside? Probably, once it's moved out of the base class, it may become irrelevant.

Reply:
I removed duplication in branches in a single commit. It also removes `'columns'` duplication. Not sure it makes code more understandable though. Could be reverted easily if needed.